Fix name mismatch between omniauth strategy and callback.

coezbek opened the pull request:
Omniauth google didn't work out of the box, because callback method was misnamed.

coolprobn commented:
Hi @coezbek, thanks for contributing.

I haven't used this gem and generator myself but going through the installation setup, it seems the controller action should be `google_oauth2 ` instead of what you have changed to, which could possibly be a typo `google_auth2 `, can you take a look?

<img width="730" alt="image" src="https://github.com/user-attachments/assets/52528a9e-df83-4931-9490-23adf105f03c">

Also apart from the code, can you address other things as per the [Contribution Guidelines](https://github.com/abhaynikam/boring_generators/blob/main/CONTRIBUTING.md#contributing-code)? I will highlight the important ones here:

[ ] Update tests to ensure generator is working correctly after the change
[ ] Manual test in the actual Rails app to ensure your changes are working as intended
[ ] Update Changelog and add an entry for this change

Note: don't forget to add yourself to contributor list at the bottom of the doc first before referencing your username in the Changelog

Here is an example PR you can take reference from https://github.com/abhaynikam/boring_generators/pull/113/files

Thanks for your time and please don't hesitate to reach out to me here in the reply if you need help with anything to get this PR to the finish line.

coezbek commented:
@coolprobn Hey thanks for the review! I fixed this again in the branch and updated the changelog.

I don't know where to begin to test the omniauth flow. The existing tests are pretty much static (https://github.com/abhaynikam/boring_generators/blob/main/test/generators/oauth/google_install_generator_test.rb)

coolprobn commented:
Hi,
I took a closer look and tested this generator in the actual app today and found some issues we need to handle and improvements we can make.

Would you mind taking a look at these so this PR can be taken to the finish line?

- [ ] In Readme, update :google_auth in instructions 2 to :google_auth2 instead 
- [ ] In Readme, update instructions 3 to include the actual URL example at the end like so -> e.g. https://boringgenerators.com/users/auth/google_oauth2/callback 
- [ ] Remove skip_before_action :verify_authenticity_token because it's not working, let's instead add the code to install the gem "omniauth-rails_csrf_protection" for handling csrf protection
  - [ ] Add tests for this 
- [ ] Update google_oauth2 action to be same as what is in the gem readme example https://github.com/zquestz/omniauth-google-oauth2?tab=readme-ov-file#devise:

```ruby
  def google_oauth2
      # You need to implement the method below in your model (e.g. app/models/user.rb)
      @user = User.from_omniauth(request.env['omniauth.auth'])

      if @user.persisted?
        flash[:notice] = I18n.t 'devise.omniauth_callbacks.success', kind: 'Google'
        sign_in_and_redirect @user, event: :authentication
      else
        session['devise.google_data'] = request.env['omniauth.auth'].except('extra') # Removing extra as it can overflow some session stores
        redirect_to new_user_registration_url, alert: @user.errors.full_messages.join("\n")
      end
  end
```

- [ ] Link @coezbek to the GitHub profile at the end of the Changelog file. If you are not seeing anything at the end then you might want to see raw version of the MD file that is editable

<img width="724" alt="image" src="https://github.com/user-attachments/assets/7ddc7807-92ed-4ebb-8346-814cb20959e4">

coolprobn commented:
Hi @coezbek 
I am planning to take over this PR so we can take it to the finish line.

I will continue from the work you have done so you are also added as a contributor.

Thank you for taking a stab at this.
